python/tskit/drawing.py

```diff
@@ -552,99 +860,197 @@ def __init__(
                 if mutation_label_attrs is not None and m in mutation_label_attrs:
                     self.mutation_label_attrs[m].update(mutation_label_attrs[m])
                 self.add_class(self.mutation_label_attrs[m], "lab")
-        self.draw()
 
-    def setup_drawing(self, style):
-        "Return an svgwrite.Drawing object for further use"
-        dwg = svgwrite.Drawing(
-            size=self.image_size, debug=True, **self.root_svg_attributes
+        # Assign coords
+        x_axis_offset = 20
+        if x_label is not None:
+            x_axis_offset += self.line_height
+        y_axis_offset = self.default_y_axis_offset
+        if y_label is not None:
+            y_axis_offset += self.line_height
+        self.set_plotbox_pad(  # To do, allow these to be set as params
+            top=10,
+            left=20 if y_label is None else y_axis_offset,
+            bottom=10 if x_label is None else (10 + x_axis_offset),
+            right=20,
         )
-        # Put all styles in a single stylesheet (required for Inkscape 0.92)
-        style = SvgTree.standard_style + ("" if style is None else style)
-        dwg.defs.add(dwg.style(style))
-        tree_class = f"tree t{self.tree.index}"
-        self.root_group = dwg.add(dwg.g(class_=tree_class))
-        return dwg
+        self.assign_y_coordinates(tree_height_scale, max_tree_height, force_root_branch)
+        self.assign_x_coordinates()
+        # Draw
+        if x_label is not None:
+            self.draw_x_axis(
+                x_axis_offset,
+                tick_positions=np.array(tree.interval),
+                tick_length_lower=self.default_tick_length,  # TODO - parameterize
+                tick_length_upper=self.default_tick_length_site,  # TODO - parameterize
+                sites=tree.sites(),
+                label=x_label,
+            )
+        if y_label is not None:
+            if y_gridlines is None:
+                y_gridlines = False if y_ticks is None else True
+            self.draw_y_axis(
+                y_axis_offset,
+                tick_length_left=self.default_tick_length,
+                tick_positions=check_ticks(y_ticks, set(self.node_height.values())),
+                label=y_label,
+                gridlines=y_gridlines,
+            )
+        self.draw_tree()
+
+    def process_mutations(self, mut_node, lower_bound, upper_bound, ignore_times=False):
+        """
+        Sort the self.node_mutations array for a given mut_node in reverse time order,
+        returning the oldest time.
+        The main complication is with UNKNOWN_TIME values: we replace those with
```

Comment:
I'm guessing there is no way to tell that a time was unknown from the SVG? Should we add some indication?

Reply:
Good point. It *should* normally be obvious, because the mutations will be spaced out evenly, but perhaps that's not enough? I could add an extra class to the mutation group, I suppose, to make it available for styling? What do you think @benjeffery ?

Reply:
A class would be good, as only people with mixed times will care.

Reply:
The CSS class is now (longwindedly) `unknown_time`, e.g. `<g class="mut m5 s2 unknown_time">`. Guess that's fine though, as it's only duplicated once per mutation.